### PR TITLE
Disable "Check for Updates" while the game is running or if all mods are removed

### LINKED ISSUE #1003: Disable the download and update mods buttons when instance is running
### Role

N/a

### Suggestion

The "Download mods" button and the "Check for Updates" buttons when viewing an instance's mods should be disabled when the instance is running, similar to the "Add file" button

### Benefit

Prevents potential problems when removing outdated mods, and potential confusion when downloaded mods aren't instantly loaded

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

It's worth noting that the "Download mods" button doesn't even work but should probably still be greyed out for clarity. The "Check for Updates" button does work but could potentially cause issues if a loaded mod is updated while the game is running.

### PR DESCRIPTION
**Pull request by Gingeh**
Fixes #1003 and #1004.

(I'm not completely confident I did the method overriding properly because I'm new to c++)

### CODE REVIEW

**Comment by DioEgizio** [comment]
You need to rebase and sign off your commits ig

**Comment by Gingeh** [comment]
Oh, no. What have I done?

File: launcher/ui/pages/instance/ModFolderPage.h [hunk 1]
```diff
@@ -53,6 +53,7 @@ class ModFolderPage : public ExternalResourcesPage {
     virtual QString helpPage() const override { return "Loader-mods"; }
 
     virtual bool shouldDisplay() const override;
+    virtual void runningStateChanged(bool running);
```
Comment (flowln) [inline launcher/ui/pages/instance/ModFolderPage.h]:
```suggestion
    void runningStateChanged(bool running) override;
```
nitpick :v

**Review by flowln — COMMENTED** [review]
only needs to fix the bot's warning, and it should be good to go!

File: launcher/ui/pages/instance/ModFolderPage.cpp [hunk 2]
```diff
@@ -84,25 +84,44 @@ ModFolderPage::ModFolderPage(BaseInstance* inst, std::shared_ptr<ModFolderModel>
         ui->actionsToolbar->insertActionAfter(ui->actionAddItem, ui->actionUpdateItem);
         connect(ui->actionUpdateItem, &QAction::triggered, this, &ModFolderPage::updateMods);
 
-        connect(ui->treeView->selectionModel(), &QItemSelectionModel::selectionChanged, this,
-                [this] { ui->actionUpdateItem->setEnabled(ui->treeView->selectionModel()->hasSelection() || !m_model->empty()); });
+        connect(ui->treeView->selectionModel(), &QItemSelectionModel::selectionChanged, this, [this] {
+            ui->actionUpdateItem->setEnabled(!(m_instance && m_instance->isRunning()) &&
+                                             (ui->treeView->selectionModel()->hasSelection() || !m_model->empty()));
```
Comment (flowln) [inline launcher/ui/pages/instance/ModFolderPage.cpp]: This expression is becoming a bit unwieldy, maybe we could separate it into another lambda, something like this:

```cpp
auto should_enable_update = [this] {
    return (!m_instance || !m_instance->isRunning()) &&
               (ui->treeView->selectionModel()->hasSelection() || !m_model->empty());
};
```

and then use it like `ui->actionUpdateItem->setEnabled(should_enable_update());`

**Review by flowln — APPROVED** [review]
thank u!

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**